### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Muckbuck**

### CODE REVIEW

File: src/BladeEngineWrapper.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        if(empty($this->getViewPaths())) {
+            throw new \Exception("Error: View paths must be defined before running init.");
+        }
+
+        //Clear cache on local instance
+        $this->maybeClearCache(); 
+
+        //Create cache path
+        $this->createComponentCachePath(); 
+        $this->addViewPath('/Users/hiq/sites/municipio/wp-content/plugins/Modularity/source/php/Module/Text/views');
```
Comment (sebastianthulin) [inline src/BladeEngineWrapper.php]: Local view path (from dev env). Should be dynamic.

File: src/BladeEngineWrapper.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        //Create new blade instance
+        $bladeEngineInstance = new BladeInstance(
+            (array) $this->getViewPaths(),
+            (string) sys_get_temp_dir() . '/global-blade-engine-cache'
```
Comment (sebastianthulin) [inline src/BladeEngineWrapper.php]: Should not use keyword global.

File: src/BladeEngineWrapper.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * Create a cache dir
+     *
+     * @return string Local path to the cache path
+     */
+    private function createComponentCachePath() : string
+    {
+
+        $cachePath = (string) sys_get_temp_dir() . '/global-blade-engine-cache';
```
Comment (sebastianthulin) [inline src/BladeEngineWrapper.php]: Should not use keyword global.

File: src/BladeEngineWrapper.php [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * Clears blade cache if in dev domain
+     *
+     * @return boolean True if cleared, false otherwise
+     */
+    private function maybeClearCache($objectPath = null)
+    {
+
+        $cachePath = (string) sys_get_temp_dir() . '/global-blade-engine-cache';
```
Comment (sebastianthulin) [inline src/BladeEngineWrapper.php]: Should not use keyword global.

File: src/BladeEngineWrapper.php [hunk 5]
```diff
@@ -0,0 +1,132 @@
+<?php
```
Comment (sebastianthulin) [inline src/BladeEngineWrapper.php:1]: Cache path is defined multiple times. Would be nice if we make a class var to define location.

File: src/BladeEngineWrapper.php (hunk 1 above)
Comment (Muckbuck) [reply src/BladeEngineWrapper.php]: Yes, forgot to remove that one.

**Pull request merged**